src/main/kotlin/org/rust/lang/core/completion/RsKeywordCompletionProvider.kt

```diff
@@ -36,3 +27,20 @@ class AddSuffixInsertionHandler(val suffix: String) : InsertHandler<LookupElemen
         EditorModificationUtil.moveCaretRelatively(context.editor, suffix.length)
     }
 }
+
+private val ALWAYS_NEEDS_SPACE = setOf("crate", "const", "enum", "extern", "fn", "impl", "let", "mod", "mut", "pub",
+    "static", "struct", "trait", "type", "unsafe", "use")
+
+
+private fun addInsertionHandler(keyword: String, builder: LookupElementBuilder, parameters: CompletionParameters): LookupElementBuilder {
+    val suffix = when (keyword) {
+        in ALWAYS_NEEDS_SPACE -> " "
+        "return" -> {
+            val fn = parameters.position.parentOfType<RsFunction>() ?: return builder
+            if (fn.retType != null) " " else ";"
```

Comment:
Inside `fn foo() -> ()` it could also be completed with `;`

Diverging functions probably also need a special case (don't suggest `return` at all), but that may be a subject for a separate PR.